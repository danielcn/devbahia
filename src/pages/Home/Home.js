import React, {Component} from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Header from '../../components/Header/Header';
import MediaCard from '../../components/MediaCard/MediaCard';

const fethedArray = [1, 2, 3, 4, 5 , 6, 7];

const mediaCards = () => fethedArray.map((index, value) => {
  return <MediaCard  key={index} />;
});

class Home extends Component {
  render () {
    return (
      <div className="Home">
        <Header/>
        <Card>
          <CardContent>
            <h1>Hello Dev Bahia</h1>
            { mediaCards() }
          </CardContent>
        </Card>
      </div>
    );
  }
}

export default Home;
