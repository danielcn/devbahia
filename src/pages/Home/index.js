import React, {Component} from 'react';

import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';

import Header from '../../components/Header';

class Home extends Component {
    render () {
        return (
            <div className="Home">
                <Header/>
                <Card>
                    <CardContent>
                        <h1>Hello Dev Bahia</h1>
                        <CardActions>
                            <Button size="small" color="primary">Learn More</Button>
                        </CardActions>
                    </CardContent>
                </Card>
            </div>
        );
    }
}

export default Home;
