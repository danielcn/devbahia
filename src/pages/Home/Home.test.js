import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

import Home from './index';

describe('Header Snapshot', () => {
    test('renders', () => {
      const component = renderer.create(
        <Home />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('should render without throwing an error', function() {
      const wrapper = shallow(<Home />);
      expect(wrapper.contains(<h1>Hello Dev Bahia</h1>)).toBe(true);
    });

});
