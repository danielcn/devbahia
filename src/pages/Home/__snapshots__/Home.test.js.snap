// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot renders 1`] = `
<div
  className="Home"
>
  <div
    className="Header"
  >
    <header
      className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-7"
    >
      <div
        className="MuiToolbar-root-37 MuiToolbar-regular-39 MuiToolbar-gutters-38"
      >
        <h6
          className="MuiTypography-root-41 MuiTypography-h6-58 MuiTypography-colorInherit-70"
        >
          Dev Bahia
        </h6>
      </div>
    </header>
  </div>
  <div
    className="MuiPaper-root-10 MuiPaper-elevation1-13 MuiPaper-rounded-11 MuiCard-root-76"
  >
    <div
      className="MuiCardContent-root-77"
    >
      <h1>
        Hello Dev Bahia
      </h1>
      <div
        className="MuiCardActions-root-78"
      >
        <button
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-textPrimary-83 MuiButton-flat-85 MuiButton-flatPrimary-86 MuiButton-sizeSmall-103 MuiCardActions-action-79"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-81"
          >
            Learn More
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
